File: 900/160A/160A.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int main(){
    int n;

    cin >> n;

    vector<int> arr(n);

    int sum = 0;

    for(int i = 0; i < n; i++){
        cin >> arr[i];
        
        sum += arr[i];
    }

    int half_val = sum / 2;

    sort(arr.begin(), arr.end());

    sum = 0;

    int num_coins = 0;

    for(int i = n - 1; i >= 0; i--){
        sum += arr[i];
        num_coins ++;

        if(sum > half_val){
            break;
        }
    }

    cout << num_coins << endl;
}

File: 900/160A/160A.py
n = int(input())
arr = list(map(int, input().split()))

sum_nums = sum(arr)
half_sum = sum_nums // 2
sorted_number = sorted(arr, reverse=True)
num_coins = 0

sum_nums = 0

for num in sorted_number:
    sum_nums += num
    num_coins += 1
    
    if sum_nums > half_sum:
        break

print(num_coins)